### PR TITLE
Upload 1200/report improvements

### PR DESCRIPTION
**Pull request by cfarmer-fearless**
Adding the following reports:
- dex-metadata-transform for overwritting the Upload ID
- dex-metadata-transform for adding the global timestamp
- dex-upload-started for when the upload starts
- dex-upload-complete for when the upload completes
- dex-metadata-transform for the v1 -> v2 manifest translation\

Updated the following reports:
- dex-file-copy

All reports for a file should look like this:
```json
{
    "upload_id": "7ea02d38f54b5291b45e5ad17979b96e",
    "data_stream_id": "dextesting",
    "data_stream_route": "testevent1",
    "reports": [
        {
            "report_id": "5197db7e-6eac-42f3-b161-f88417193000",
            "stage_name": "dex-metadata-transform",
            "timestamp": "2024-05-20T14:47:09.860Z",
            "content": {
                "schema_version": "1.0",
                "field": "ID",
                "action": "update",
                "schema_name": "metadata-transform",
                "value": "7ea02d38f54b5291b45e5ad17979b96e"
            }
        },
        {
            "report_id": "698362d9-c98a-4b5c-96a2-c9c01488eeda",
            "stage_name": "dex-metadata-transform",
            "timestamp": "2024-05-20T14:47:09.920Z",
            "content": {
                "schema_version": "1.0",
                "field": "dex_ingest_datetime",
                "action": "append",
                "schema_name": "metadata-transform",
                "value": "2024-05-20T14:47:09Z"
            }
        },
        {
            "report_id": "0f7b5cb6-11a9-4af2-b689-ad7cc99a8775",
            "stage_name": "dex-metadata-verify",
            "timestamp": "2024-05-20T14:47:10.020Z",
            "content": {
                "schema_version": "0.0.1",
                "metadata": {
                    "dex_ingest_datetime": "2024-05-20T14:47:09Z",
                    "filename": "really-small-file.txt",
                    "filetype": "text/plain",
                    "message_type": "ELR",
                    "meta_data_stream_id": "dextesting",
                    "meta_ext_entity": "DD2",
                    "meta_ext_event": "testevent1",
                    "meta_ext_file_timestamp": "",
                    "meta_ext_filename": "really-small-file.txt",
                    "meta_ext_filestatus": "Succeeded",
                    "meta_ext_objectkey": "a49433ce-742b-42f3-8c40-7b24fa0bdb27",
                    "meta_ext_source": "IZGW",
                    "meta_ext_sourceversion": "V2022-12-31",
                    "meta_ext_submissionperiod": "1",
                    "meta_ext_uploadid": "98112743",
                    "meta_organization": "IL",
                    "meta_program": "test-program",
                    "meta_username": "manual test",
                    "original_filename": "really-small-file",
                    "reporting_jurisdiction": "daart reporting jurisdiction",
                    "route": "DAART"
                },
                "filename": "really-small-file.txt",
                "schema_name": "dex-metadata-verify"
            }
        },
        {
            "report_id": "a320fe63-e023-4b2b-8f88-a3e8a9dbde91",
            "stage_name": "dex-upload-started",
            "timestamp": "2024-05-20T14:47:10.109Z",
            "content": {
                "schema_version": "1.0",
                "schema_name": "dex-upload-started",
                "status": "success"
            }
        },
        {
            "report_id": "c1be1641-825d-4066-93b7-1d2357c80f3c",
            "stage_name": "dex-metadata-transform",
            "timestamp": "2024-05-20T14:47:22.566Z",
            "content": {
                "schema_version": "1.0",
                "transforms": {
                    "action": "add",
                    "items": [
                        {
                            "field": "sender_id",
                            "value": "IZGW"
                        },
                        {
                            "field": "data_producer_id",
                            "value": ""
                        },
                        {
                            "field": "jurisdiction",
                            "value": "DD2"
                        },
                        {
                            "field": "received_filename",
                            "value": "really-small-file.txt"
                        },
                        {
                            "field": "data_stream_id",
                            "value": "dextesting"
                        },
                        {
                            "field": "data_stream_route",
                            "value": "testevent1"
                        },
                        {
                            "field": "version",
                            "value": "2.0"
                        }
                    ]
                },
                "schema_name": "dex-metadata-transform"
            }
        },
        {
            "report_id": "254161f3-5d26-4a1c-91ab-45ba8eeac788",
            "stage_name": "dex-file-copy",
            "timestamp": "2024-05-20T14:47:23.325Z",
            "content": {
                "result": "success",
                "schema_version": "0.0.1",
                "destination": "routing",
                "schema_name": "dex-file-copy"
            }
        },
        {
            "report_id": "d14f427d-bba8-49c5-92a2-5c7e77039dde",
            "stage_name": "dex-file-copy",
            "timestamp": "2024-05-20T14:47:23.605Z",
            "content": {
                "result": "success",
                "schema_version": "0.0.1",
                "destination": "edav",
                "schema_name": "dex-file-copy"
            }
        },
        {
            "report_id": "da3c1029-6fc3-4766-98e4-c5347e2c5bc8",
            "stage_name": "dex-upload-complete",
            "timestamp": "2024-05-20T14:47:25.785Z",
            "content": {
                "schema_version": "1.0",
                "schema_name": "dex-upload-complete",
                "status": "success"
            }
        },
        {
            "report_id": "51237392-9d63-452f-8873-42b8187dcd27",
            "stage_name": "dex-upload-status",
            "timestamp": "2024-05-20T14:47:25.927Z",
            "content": {
                "schema_version": "1.0",
                "filename": "really-small-file.txt",
                "offset": "800000",
                "size": "1080320",
                "schema_name": "upload",
                "tguid": "7ea02d38f54b5291b45e5ad17979b96e"
            }
        }
    ]
}
```

### CODE REVIEW

File: upload-server/internal/metadata/metadata.go [hunk 1]
```diff
@@ -218,18 +218,52 @@ func (v *SenderManifestVerification) Verify(event handler.HookEvent, resp hooks.
...
-func WithUploadID(event handler.HookEvent, resp hooks.HookResponse) (hooks.HookResponse, error) {
+func (v *HookEventHandler) WithUploadID(event handler.HookEvent, resp hooks.HookResponse) (hooks.HookResponse, error) {
 
 	tuid := Uid()
 	resp.ChangeFileInfo.ID = tuid
 
 	logger.Info("Generated UUID", "UUID", tuid)
 
+	content := &models.MetaDataTransformContent{
+		ReportContent: models.ReportContent{
```
Comment (tedtally) [inline upload-server/internal/metadata/metadata.go:229]: Are we using camel case in Go?

File: upload-processor/BulkFileUploadFunctionApp/Services/IBulkUploadSvcBusClient.cs
[the remark this answers is not in the document]
Comment (cfarmer-fearless) [reply upload-processor/BulkFileUploadFunctionApp/Services/IBulkUploadSvcBusClient.cs:1]: Which line are you referring to?

File: upload-server/internal/metadata/metadata.go (hunk 1 above)
Comment (cfarmer-fearless) [reply upload-server/internal/metadata/metadata.go:229]: Struct names are capital case in Go

**Review by whytheplatypus — CHANGES_REQUESTED** [review]
we need to be able to differentiate the stage an event is happening in. In this case the transform reports from the processor and the upload stages look the same. Can we use stage name to consistently say what step in the process this is in? Or add a new field for now like `service` so we know which one sent it out?

**Comment by cfarmer-fearless** [comment]
Created follow on ticket for this work: https://cdc-dexops.atlassian.net/browse/UPLOAD-1352

**Review by whytheplatypus — APPROVED** [review]

**Pull request merged**